ehsm_kms_service/secret_manager_apis.js

```diff
@@ -265,9 +265,119 @@ const createSecret = async (res, appid, payload, DB) => {
     }
 }
 
+/**
+ * stores the secret value of a new version into a secret object.
+ * @param {Object} res : response
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database control
+ * @param {Object} payload
+ *          ==> {r}secretName(String [1~64]) : The name of the secret. eg. 'secretName01'
+ *          ==> {r}secretData(String [1~4096]) : The value of the secret.                                             eg. '30d'
+ * @returns
+ */
+const putSecretValue = async (res, appid, payload, DB) => {
+    try {
+        //get and check Param in payload
+        let secretName = getParam_String(payload, 'secretName')
+        let secretData = getParam_String(payload, 'secretData')
+        const createTime = new Date().getTime()
+        let sm_masterKey
+        let encryptionKeyId
+        let versionId
+        if (!checkStringParam(secretData, true, SECRETDATA_LENGTH_MAX)) {
+            res.send(_result(400, `secretData cannot be empty, must be string and length not more than ${SECRETDATA_LENGTH_MAX}`))
+            return
+        }
+        if (!checkStringParam(secretName, true, SECRETNAME_LENGTH_MAX)) {
+            res.send(_result(400, `secretName cannot be empty, must be string and length not more than ${SECRETNAME_LENGTH_MAX}`))
+            return
+        }
+
+        //
+        //query SM_Masterkey and encrypt secretData
+        const query_secertName = {
+            selector: {
+                appid,
+                secretName,
+            },
+            limit: 1
+        }
+        const secret_metadata_res = await DB.partitionedFind('secret_metadata', query_secertName)
+        if (secret_metadata_res.docs.length == 0) {
+            res.send(_result(400, 'can not find secretName'))
+            return
+        } else {
+            if(secret_metadata_res.docs[0].encryptionKeyId == undefined) {
+                encryptionKeyId = ""
+            }
+            else {
+                encryptionKeyId = secret_metadata_res.docs[0].encryptionKeyId
+            }
+        }
+        retCMK = await get_SM_Masterkey(DB, appid, encryptionKeyId)
+        if (retCMK.isFind) {
+            sm_masterKey = retCMK.sm_masterKey
+        } else {
+            res.send(_result(400, retCMK.msg))
+            return
+        }
+        const apikey_encrypt_res = napi_result(cryptographic_apis.Encrypt, res, [sm_masterKey, secretData, ''])
+        const { ciphertext } = apikey_encrypt_res.result
```

Comment:
will check any error returned from Encrypt?

Reply:
will catch in line 370